plugins/out_stackdriver/stackdriver.c

```diff
@@ -546,22 +611,41 @@ static int stackdriver_format(const void *data, size_t bytes,
          *  "timestamp": "..."
          * }
          */
+        
+        /*  Parse jsonPayload and extract operation first*/
+        operation_id = flb_sds_create("");
+        operation_producer = flb_sds_create("");
+        operation_existed = extract_operation(&operation_id, &operation_producer,
+                              &operation_first, &operation_last, obj);
+
+        entry_size += calculate_spec_fields(operation_existed);
+
         if (ctx->severity_key
             && get_severity_level(&severity, obj, ctx->severity_key) == 0) {
             /* additional field for severity */
-            msgpack_pack_map(&mp_pck, 4);
+            entry_size += 1;
+            msgpack_pack_map(&mp_pck, entry_size);
             msgpack_pack_str(&mp_pck, 8);
             msgpack_pack_str_body(&mp_pck, "severity", 8);
             msgpack_pack_int(&mp_pck, severity);
         }
         else {
-            msgpack_pack_map(&mp_pck, 3);
+            msgpack_pack_map(&mp_pck, entry_size);
+        }
+
+        /* Add operation field into the log entry */
+        if (operation_existed == true) {
+            add_operation_field(&operation_id, &operation_producer,
+                                &operation_first, &operation_last, &mp_pck);
         }
+        
+        /* print out the jsonPayload*/
+        print_json_payload(JSONPAYLOAD_WRITE_PATH, obj);
 
         /* jsonPayload */
         msgpack_pack_str(&mp_pck, 11);
         msgpack_pack_str_body(&mp_pck, "jsonPayload", 11);
-        msgpack_pack_object(&mp_pck, *obj);
+        update_json_payload(operation_existed, &mp_pck, obj);
```

Comment:
What do you think calling it `pack_jsonpayload`? As we are packing the object into the value of `jsonPayload` based on other logics (e.g. operation_existed here).

Reply:
Done.